# **(ADD THE NOTEBOOK NAME HERE)**

## Objectives

* Write your notebook objective here, for example, "Fetch data from Kaggle and save as raw data", or "engineer features for modelling"

## Inputs

* Write down which data or information you need to run the notebook 

## Outputs

* Write here which files, code or artefacts you generate by the end of the notebook 

## Additional Comments

* If you have any additional comments that don't fit in the previous bullets, please state them here. 



---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [12]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\leha_\\Documents\\vscode-projects\\football-players-data\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [13]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [14]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\leha_\\Documents\\vscode-projects\\football-players-data'

# Section 1

Section 1 content

In [1]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
sns.set_style("whitegrid")


---

# Section 2

ETL Process (Extract, Transfor, Load) - Start with loading the raw dataset.

In [15]:
# check the files in the current directory
print(os.listdir(current_dir))



['.git', '.github', '.gitignore', '.python-version', '.slugignore', '.venv', 'DataSet', 'jupyter_notebooks', 'Procfile', 'README.md', 'requirements.txt', 'setup.sh']


In [18]:
# load raw data
df = pd.read_csv('DataSet/RawData/fifa_players.csv')

# check the first 5 rows of the dataframe
df.head()

,name,full_name,birth_date,age,height_cm,weight_kgs,positions,nationality,overall_rating,potential,...,long_shots,aggression,interceptions,positioning,vision,penalties,composure,marking,standing_tackle,sliding_tackle
0,L. Messi,Lionel Andrés Messi Cuccittini,6/24/1987,31,170.18,72.1,"CF,RW,ST",Argentina,94,94,...,94,48,22,94,94,75,96,33,28,26
1,C. Eriksen,Christian Dannemann Eriksen,2/14/1992,27,154.94,76.2,"CAM,RM,CM",Denmark,88,89,...,89,46,56,84,91,67,88,59,57,22
2,P. Pogba,Paul Pogba,3/15/1993,25,190.50,83.9,"CM,CAM",France,88,91,...,82,78,64,82,88,82,87,63,67,67
3,L. Insigne,Lorenzo Insigne,6/4/1991,27,162.56,59.0,"LW,ST",Italy,88,88,...,84,34,26,83,87,61,83,51,24,22
4,K. Koulibaly,Kalidou Koulibaly,6/20/1991,27,187.96,88.9,CB,Senegal,88,91,...,15,87,88,24,49,33,80,91,88,87


In [19]:
# use tail to check the last 5 rows of the dataframe
df.tail()

,name,full_name,birth_date,age,height_cm,weight_kgs,positions,nationality,overall_rating,potential,...,long_shots,aggression,interceptions,positioning,vision,penalties,composure,marking,standing_tackle,sliding_tackle
17949,R. McKenzie,Rory McKenzie,10/7/1993,25,175.26,74.8,"RM,CAM,CM",Scotland,67,70,...,54,69,41,60,64,63,56,40,20,18
17950,M. Sipľak,Michal Sipľak,2/2/1996,23,182.88,79.8,LB,Slovakia,59,67,...,22,62,55,42,39,32,52,53,64,60
17951,J. Bekkema,Jan Bekkema,4/9/1996,22,185.42,89.8,GK,Netherlands,59,67,...,9,27,10,5,25,16,47,9,12,13
17952,A. Al Yami,Abdulrahman Al Yami,6/19/1997,21,175.26,64.9,"ST,LM",Saudi Arabia,59,71,...,58,38,15,54,52,50,53,16,18,17
17953,Júnior Brumado,José Francisco dos Santos Júnior,5/15/1999,19,190.50,79.8,ST,Brazil,59,75,...,53,67,20,53,49,52,45,23,31,21


In [20]:
# Use info to check the data types and missing values
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17954 entries, 0 to 17953
Data columns (total 51 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   name                           17954 non-null  object 
 1   full_name                      17954 non-null  object 
 2   birth_date                     17954 non-null  object 
 3   age                            17954 non-null  int64  
 4   height_cm                      17954 non-null  float64
 5   weight_kgs                     17954 non-null  float64
 6   positions                      17954 non-null  object 
 7   nationality                    17954 non-null  object 
 8   overall_rating                 17954 non-null  int64  
 9   potential                      17954 non-null  int64  
 10  value_euro                     17699 non-null  float64
 11  wage_euro                      17708 non-null  float64
 12  preferred_foot                 17954 non-null 

The above shows the information of the dataset. There is a total of 51 columns in the dataset and 17,954 rows. It also shows the column names and the data types. The non-null count of each column helps to identify any missing values. I'll confirm by using df.isnull().sum() to check the columns for further analysis.

Cleaning the data means fixing the bad data in our dataset. So this could be empty cells, data in the wrong format, wrong data and duplicates.

In [21]:
# check sum of missing values in each column
df.isnull().sum()

name                                 0
full_name                            0
birth_date                           0
age                                  0
height_cm                            0
weight_kgs                           0
positions                            0
nationality                          0
overall_rating                       0
potential                            0
value_euro                         255
wage_euro                          246
preferred_foot                       0
international_reputation(1-5)        0
weak_foot(1-5)                       0
skill_moves(1-5)                     0
body_type                            0
release_clause_euro               1837
national_team                    17097
national_rating                  17097
national_team_position           17097
national_jersey_number           17097
crossing                             0
finishing                            0
heading_accuracy                     0
short_passing            

The value euro, 255 and wage euro 246 and release clause euro 1,837, national team and national rating, national team position and national jersey number has 17097 rows.

In [22]:
# remove missing values
df.dropna(inplace=True)


We remove all the rows with the missing value using the df.dropna method.

In [23]:
# describe the dataframe to check the summary statistics
df.describe()


,age,height_cm,weight_kgs,overall_rating,potential,value_euro,wage_euro,international_reputation(1-5),weak_foot(1-5),skill_moves(1-5),...,long_shots,aggression,interceptions,positioning,vision,penalties,composure,marking,standing_tackle,sliding_tackle
count,789.000000,789.000000,789.000000,789.000000,789.000000,7.890000e+02,789.000000,789.000000,789.000000,789.000000,...,789.000000,789.000000,789.000000,789.000000,789.000000,789.000000,789.000000,789.000000,789.000000,789.000000
mean,26.681876,176.670038,77.304309,75.970849,78.887199,1.290264e+07,44812.420786,1.612167,3.152091,2.749049,...,55.692015,63.278834,54.095057,57.787072,63.636248,53.514575,69.627376,53.558935,53.721166,50.911280
std,3.631705,14.160303,7.355305,6.091345,6.116359,1.642863e+07,62801.821871,0.875008,0.755478,0.964017,...,21.784424,17.914654,22.798217,22.536666,13.971445,17.854285,10.879383,22.680257,24.612960,24.628452
min,18.000000,152.400000,59.000000,58.000000,62.000000,2.200000e+05,1000.000000,1.000000,1.000000,1.000000,...,5.000000,11.000000,7.000000,4.000000,12.000000,10.000000,23.000000,7.000000,9.000000,8.000000
25%,24.000000,170.180000,72.100000,72.000000,74.000000,2.800000e+06,8000.000000,1.000000,3.000000,2.000000,...,43.000000,54.000000,32.000000,47.000000,56.000000,42.000000,64.000000,34.000000,30.000000,25.000000
50%,26.000000,182.880000,77.100000,75.000000,79.000000,7.000000e+06,22000.000000,1.000000,3.000000,3.000000,...,63.000000,68.000000,63.000000,65.000000,66.000000,56.000000,71.000000,62.000000,64.000000,61.000000
75%,29.000000,187.960000,82.100000,80.000000,83.000000,1.600000e+07,52000.000000,2.000000,4.000000,3.000000,...,72.000000,76.000000,74.000000,74.000000,74.000000,67.000000,77.000000,72.000000,75.000000,72.000000
max,39.000000,203.200000,99.800000,94.000000,95.000000,1.105000e+08,565000.000000,5.000000,5.000000,5.000000,...,94.000000,94.000000,92.000000,95.000000,94.000000,90.000000,96.000000,94.000000,93.000000,90.000000


In [24]:
#check the shape of the dataframe
df.shape

(789, 51)

In [25]:
#check the missing values again to confirm that they are removed
df.isnull().sum()

name                             0
full_name                        0
birth_date                       0
age                              0
height_cm                        0
weight_kgs                       0
positions                        0
nationality                      0
overall_rating                   0
potential                        0
value_euro                       0
wage_euro                        0
preferred_foot                   0
international_reputation(1-5)    0
weak_foot(1-5)                   0
skill_moves(1-5)                 0
body_type                        0
release_clause_euro              0
national_team                    0
national_rating                  0
national_team_position           0
national_jersey_number           0
crossing                         0
finishing                        0
heading_accuracy                 0
short_passing                    0
volleys                          0
dribbling                        0
curve               

---

NOTE

* You may add as many sections as you want, as long as it supports your project workflow.
* All notebook's cells should be run top-down (you can't create a dynamic wherein a given point you need to go back to a previous cell to execute some task, like go back to a previous cell and refresh a variable content)

---

# Push files to Repo

* In cases where you don't need to push files to Repo, you may replace this section with "Conclusions and Next Steps" and state your conclusions and next steps.

In [ ]:
import os
try:
  # create your folder here
  # os.makedirs(name='')
except Exception as e:
  print(e)
